### ISSUE TITLE
Incorrect link_name is generated for extern "C" functions for 32 bit linux

### ISSUE THREAD

**Issue by chengchangwu** [rust-lang-nursery/rust-bindgen#904]
Like #728, but for 32 bit Linux.

### Input C/C++ Header

```C
struct ec_sdo_request;
typedef struct ec_sdo_request ec_sdo_request_t;
int *ecrt_sdo_request_data(
        ec_sdo_request_t *req /**< SDO request. */
        );
```

### Bindgen Invocation

```
$ bindgen input.h
```

### Actual Results

```rust
#[repr(C)]
#[derive(Debug, Copy, Clone)]
pub struct ec_sdo_request {
    _unused: [u8; 0],
}
pub type ec_sdo_request_t = ec_sdo_request;
extern "C" {
    #[link_name = "_Z21ecrt_sdo_request_data"]
    pub fn ecrt_sdo_request_data(req: *mut ec_sdo_request_t)
     -> *mut ::std::os::raw::c_int;
}
```
### Expected Results

```Rust
#[repr(C)]
#[derive(Debug, Copy, Clone)]
pub struct ec_sdo_request {
    _unused: [u8; 0],
}
pub type ec_sdo_request_t = ec_sdo_request;
extern "C" {
    pub fn ecrt_sdo_request_data(req: *mut ec_sdo_request_t)
     -> *mut ::std::os::raw::c_int;
}
```

**Comment by emilio** [comment]
Which clang version is this?

**Comment by emilio** [comment]
This sounds like https://github.com/rust-lang-nursery/rust-bindgen/issues/528

**Comment by chengchangwu** [comment]
I've clang-3.8, clang-3.9 and and old libclang1-3.6 installed. After removing clang-3.8 and old libclang1-3.6, the problem is fixed.

Thanks.

**Comment by emilio** [comment]
Closed as a dupe of #528 then, thanks for checking!

There's a workaround explained there.
